pkg/yurtmanager/webhook/yurtappoverrider/v1alpha1/yurtappoverrider_validation.go

```diff
@@ -0,0 +1,109 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an AS IS BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openyurtio/openyurt/pkg/apis/apps/v1alpha1"
+)
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type.
+func (webhook *YurtAppOverriderHandler) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	overrider, ok := obj.(*v1alpha1.YurtAppOverrider)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a YurtAppOverrider but got a %T", obj))
+	}
+
+	// validate
+	if err := webhook.validateOneToOne(ctx, overrider); err != nil {
+		return err
+	}
+	if err := webhook.validateStar(overrider); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type.
+func (webhook *YurtAppOverriderHandler) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) error {
+	_, ok := newObj.(*v1alpha1.YurtAppOverrider)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a YurtAppOverrider but got a %T", newObj))
+	}
+	newOverrider, ok := oldObj.(*v1alpha1.YurtAppOverrider)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a YurtAppOverrider} but got a %T", oldObj))
+	}
+
+	// validate
+	if err := webhook.validateOneToOne(ctx, newOverrider); err != nil {
+		return err
+	}
+	if err := webhook.validateStar(newOverrider); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type.
+func (webhook *YurtAppOverriderHandler) ValidateDelete(_ context.Context, obj runtime.Object) error {
+	_, ok := obj.(*v1alpha1.YurtAppOverrider)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a YurtAppOverrider but got a %T", obj))
+	}
+	// validate
+	return nil
+}
+
+// YurtAppOverrider and YurtAppSet are one-to-one relationship
+func (webhook *YurtAppOverriderHandler) validateOneToOne(ctx context.Context, yurtAppOverrider *v1alpha1.YurtAppOverrider) error {
+	app := yurtAppOverrider.Subject
+	var allOverriderList v1alpha1.YurtAppOverriderList
+	if err := webhook.Client.List(ctx, &allOverriderList, client.InNamespace(yurtAppOverrider.Namespace)); err != nil {
+		klog.Info("error in listing YurtAppOverrider")
+		return err
+	}
+	var overriderList = v1alpha1.YurtAppOverriderList{}
+	for _, overrider := range overriderList.Items {
+		if overrider.Subject.Kind == app.Kind && overrider.Name == app.Name && overrider.APIVersion == app.APIVersion {
```

Comment:
if overrider.Subject.Kind == app.Kind && overrider.Name == app.Name && overrider.APIVersion == app.APIVersion --> if overrider.Name == app.Name

Reply:
What if yurtappset and yurtappdaemon have the same name? Is it necessary to check kind?